app/models/opponent.rb

```diff
@@ -36,6 +36,26 @@ class Opponent < ApplicationRecord
     interested_party: 'Interested party'
   }.freeze
 
+  def self.dummy_opponent
+    Opponent.find_by(other_party_id: 'OPPONENT_00000001') || create_dummy_opponent
```

Comment:
I suggest one of these:
```ruby
    Opponent.create_with(
      title: 'MR.',
      first_name: 'Dummy',
      surname: 'Opponent',
      date_of_birth: Date.new(2015, 1, 1),
      relationship_to_case: 'OPP',
      relationship_to_client: 'EX_SPOUSE',
      opponent_type: 'PERSON',
      opp_relationship_to_case: 'Opponent',
      opp_relationship_to_client: 'Ex Husband/ Wife',
      child: false
    ).find_or_create_by!(other_party_id: 'OPPONENT_00000001')
```
```ruby
    Opponent.find_or_create_by!(other_party_id: 'OPPONENT_00000001') do |opponent|
      opponent.title = 'MR.'
      opponent.first_name = 'Dummy'
      opponent.surname = 'Opponent'
      opponent.date_of_birth = Date.new(2015, 1, 1)
      opponent.relationship_to_case = 'OPP'
      opponent.relationship_to_client = 'EX_SPOUSE'
      opponent.opponent_type = 'PERSON'
      opponent.opp_relationship_to_case = 'Opponent'
      opponent.opp_relationship_to_client = 'Ex Husband/ Wife'
      opponent.child = false
    end
```
Both are equivalent (see https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by)

Reply:
Yep, good point, thanks.